core/sql/cli/Statement.cpp

```diff
@@ -2298,6 +2298,10 @@ RETCODE Statement::resolveNames(LateNameInfoList * lnil,
 	      if (retcode != ex_expr::EXPR_OK)
 		{
 		  diagsArea << DgSqlCode(-EXE_CONVERT_STRING_ERROR);
+                  char hexstr[MAX_OFFENDING_SOURCE_DATA_DISPLAY_LEN];
```

Comment:
Need to allow room for a trailing null. You'll have to declare one more byte here than you pass as the second parameter to stringToHex. Alternatively, change the stringToHex function to allow for room (for example, change the statement "outLen = (outLen / 2)" to "outLen = (outLen / 2) - 1"